### DISCUSSION TITLE
pictures or diagrams of data and query patterns ?

### DISCUSSION THREAD

**Discussion by denis-bz** [ddotta/cookbook-rpolars#28] [Q&A] [answered]
Hi Damien
  (not an issue, just a question, please move to Discussions)
I think I saw pictures or diagrams of data and query patterns, little trees ?
under polars someplace, but can't find it again  and don't know the right terms.
I'm not a database guy, but there are so many different ways of storing and querying data
that little pictures might be fun / help intuition.
Does that ring a bell  or am I way off ?
Thanks, cheers
  -- denis

**Comment by ddotta** [comment]
Hello @denis-bz,
Sorry for the delay in replying.
The images or diagrams you're talking about don't ring a bell and I've never come across them.
Maybe other Polars users will have an idea?
Maybe @eitsupi or @etiennebacher?

**Reply by denis-bz** [reply]
Hi Damien,
  found it, wasn't polars after all: https://github.com/tableau/query-graphs
Thanks, cheers

**Reply by etiennebacher** [reply]
Polars can create query trees: https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.meta.tree_format.html

This has been improved in the development version, see here for an example: https://github.com/pola-rs/polars/pull/11176

AFAIK this is not available in the R version for now

**Reply by etiennebacher** [reply, marked as answer]
Actually the example above only creates trees that detail polars expressions.

For query trees, use `show_graph`: https://pola-rs.github.io/polars/py-polars/html/reference/lazyframe/api/polars.LazyFrame.show_graph.html

**Comment by ddotta** [comment]
Thank you both for this information, I close the issue

**Comment by denis-bz** [comment]
Folks,
  would a gallery of pictures of a dozen different query trees be informative, fun ?
Seems to me there's quite a gap between users' views of queries, and developers' views;
could little pictures, easy to collect and send around, help close the gap ?
Ideally they could be used both to analyze and test --
  real queries --> Qpic, query pics --> testbench

Does that make sense, what do you think ?
